src/http/http_server_impl.h

```diff
@@ -0,0 +1,47 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <dsn/tool-api/rpc_message.h>
+#include <dsn/cpp/serverlet.h>
+#include <dsn/http/http_server.h>
+
+namespace dsn {
+
+class http_server : public serverlet<http_server>
```

Comment:
If it's a singleton, why did you not make it inheritance from `utils::singleton`?

Reply:
I would like to use the least possible code. There're ways to implement one semantic in c++. I just choose whatever that can work.